lib/streamlit/delta_generator.py

```diff
@@ -396,18 +396,23 @@ def beta_columns(self, weights):
         >>> col3.checkbox('Good to go~')
 
         """
+        weights_exception = StreamlitAPIException(
+            "The input argument to st.beta_columns must be either a "
+            + "positive integer or a list of numeric weights. "
+            + "See [documentation](https://docs.streamlit.io/en/stable/api.html#streamlit.beta_columns) "
+            + "for more information."
+        )
 
         if isinstance(weights, int):
             if weights <= 0:
-                raise StreamlitAPIException("You have to create at least one column!")
-            if weights == 1:
-                raise StreamlitAPIException(
-                    "Instead of creating only one column, use st.beta_container."
-                )
+                raise weights_exception
```

Comment:
Any reason why it's needed? Isn't line 413 going to catch it?

Reply:
Mm, this is the case when weights is a single int (not a list), and we're supposed to use that to interpret how many columns there are.

Line 413 deals with tuples/lists of column weights, but would not work if e.g. the input was a single int, since that's not iterable.

Hm, now that I think about it, it would probably raise a rather obtuse error in that case. We could try wrapping the entire thing with a try-catch so that StreamlitAPIException is always raised, rather than NotIterable or whatever would normally happen -- WDYT?

Reply:
I guess I am interpreting lines 406-411 as normalizing the input to be a tuple. After line 411, `weights` is a tuple now and so line 413 handles the validation check. Maybe I'm missing something.

Reply:
Yeah, that's all correct. So the question is, what's going to happen if the user tries `st.beta_column(-2)` for some reason?

Without the checks in 407-408, we'd actually proceed through the entire code without showing an error (I think), since the normalization step actually doesn't fail for negative weights:

```
# In python, multiplying a tuple by a negative weight gives an empty list
>>> (1,) * -2
()
```

Reply:
Ah! This was a fail on my part. I thought python was multiplying the weights in by to produce `(-2)`. Testing it out, I see the the multiplication in increasing the size of the tuple! Of course the comment says it too, so I'm boiling it down to me being an idiot. 😊 

You could check if `len(weights) == 0 or any(....)` if you want a catchall, but this is also fine. I was just hoping to simplify the error logic a little bit.

Reply:
Good point! I've gone ahead and changed it to the catchall version =)